javascript/src/seedsource-ui/components/Popup.tsx

```diff
@@ -0,0 +1,289 @@
+        .then(json => {
+          const { results } = json
+          const validRegions = results.map((region: any) => region.name)
+
+          const region = validRegions.length ? validRegions[0] : null
+          this.setState({ region })
+          return region
+        })
+        .then(regionName => {
+          if (regionName !== null) {
+            // Set elevation at point
+            const url = `/arcgis/rest/services/${regionName}_dem/MapServer/identify/?${io.urlEncode({
+              f: 'json',
+              tolerance: '2',
+              imageDisplay: '1600,1031,96',
+              geometryType: 'esriGeometryPoint',
+              mapExtent: '0,0,0,0',
+              geometry: JSON.stringify({ x: point.x, y: point.y }),
+            })}`
+
+            io.get(url)
+              .then(response => response.json())
+              .then(json => {
+                const { results } = json
+                let value = null
+
+                if (results.length) {
+                  value = results[0].attributes['Pixel value']
+                }
+
+                if (Number.isNaN(value)) {
+                  value = null
+                }
+
+                this.setState({ elevation: value })
+              })
+
+            this.updateVariables(selectedVariables)
+
+            // Find seedzones at point
+            const zonesUrl = `${config.apiRoot}seedzones/?${io.urlEncode({ point: `${point.x},${point.y}` })}`
+
+            this.setState({ zonesFetching: this.state.zonesFetching + 1 })
+            io.get(zonesUrl)
+              .then(response => response.json())
+              .then((json: any) => {
+                const zones = json.results.map((zone: any) => ({
+                  id: zone.zone_uid,
+                  name: zone.name,
+                  elevation_band: zone.elevation_band,
+                }))
+                this.setState({
+                  zones,
+                })
+              })
+              .finally(() => this.setState({ zonesFetching: this.state.zonesFetching - 1 }))
+          }
+        })
+    }
+  }
+
+  componentDidMount() {
+    this.props.map.on('popupclose', () => {
+      this.props.onClose ? this.props.onClose() : ''
+    })
+    this.setState({
+      popup: L.popup({ closeOnClick: false }).setLatLng([this.props.point.y, this.props.point.x]).addTo(this.props.map),
+    })
+    this.updateData()
+  }
+
+  componentDidUpdate(prevProps: Readonly<PopupProps>) {
+    if (JSON.stringify(prevProps.point) !== JSON.stringify(this.props.point)) {
+      this.updateData()
+    } else if (JSON.stringify(prevProps.selectedVariables) !== JSON.stringify(this.props.selectedVariables)) {
+      this.setState({ variables: [] })
+      this.updateVariables(this.props.selectedVariables)
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.map.closePopup(this.state.popup)
+    this.setState({ popup: null })
+  }
+
+  render() {
+    setTimeout(() => {
+      if (this.state.popup) {
```

Comment:
You can remove this condition after `popup` is initialized in the constructor and never reset to `null`.